src/tileset.rs

```diff
@@ -67,6 +94,18 @@ impl Tileset {
             },
         });
 
+        let columns = match columns {
+            Some(col) => col,
+            None => match &image {
+                None => {
+                    return Err(TiledError::MalformedAttributes(
+                        "No <image> nor columns attribute in <tileset>".to_string(),
+                    ))
+                }
+                Some(image) => image.width as u32 / width,
```

Comment:
I guess this fallback should take spacing and margin into account? So `(image.width - margin + spacing) / (tile_width + spacing)`.